spec/truffle/interop/java_string_spec.rb

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+# 
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+require_relative '../../ruby/spec_helper'
+
+describe "Java strings" do
+  
+  it "are boxed" do
+    Truffle::Interop.boxed?(Truffle::Interop.to_java_string('test')).should be_true
+  end
+  
+  it "unbox to Ruby strings" do
+    Truffle::Interop.unbox(Truffle::Interop.to_java_string('test')).should == 'test'
+  end
```

Comment:
UNBOX on something that is already primitive passes it through unchanged. I did that so that you can UNBOX something if you aren't sure if it's a primitive or not yet.

Reply:
But this effectively converts/boxes a Java String to a Ruby String.

Reply:
It unboxes the string... but then all interop operations then convert any Java strings returned from interop to Ruby strings.